File: Dot examples/ASP Book Examples/Book Examples/Chapter29/VB/App_Code/ContentRotator.vb
Imports System
Imports System.Web.UI
Imports System.Web.UI.WebControls

Namespace myControls
    <ParseChildren(False)> _
    Public Class ContentRotator
        Inherits WebControl

        Protected Overrides Sub AddParsedSubObject(ByVal obj As Object)
            If TypeOf obj Is Content Then
                MyBase.AddParsedSubObject(obj)
            End If
        End Sub

        Protected Overrides Sub RenderContents(ByVal writer As HtmlTextWriter)
            Dim rnd As Random = New Random()
            Dim index As Integer = rnd.Next(Me.Controls.Count)
            Me.Controls(index).RenderControl(writer)
        End Sub
    End Class

    Public Class Content
        Inherits Control
    End Class
End Namespace

File: Dot examples/ASP Book Examples/Book Examples/Chapter29/CS/App_Code/ContentRotator.cs
using System;
using System.Web.UI;
using System.Web.UI.WebControls;

namespace myControls
{
    [ParseChildren(false)]
    public class ContentRotator : WebControl
    {

        protected override void AddParsedSubObject(object obj)
        {
            if (obj is Content)
                base.AddParsedSubObject(obj);
        }

        protected override void RenderContents(HtmlTextWriter writer)
        {
            Random rnd = new Random();
            int index = rnd.Next(this.Controls.Count);
            this.Controls[index].RenderControl(writer);
        }
    }

    public class Content : Control
    {


    }
}